test-suite/Network/CloudSeeder/Test/Stubs.hs

```diff
@@ -34,15 +33,25 @@ stubCommandLineT fake (ArgumentsT x) = runReaderT x fake
 instance Monad m => MonadCLI (ArgumentsT m) where
   getArgs = ArgumentsT $ do
     input <- ask
-    return $ consume parseArguments $ take 3 input
+    consume parseArguments $ take 3 input
 
   getOptions pSpecs = ArgumentsT $ do
     input <- ask
-    let x = execParserPure defaultPrefs (parseOptions pSpecs) input
-    return $ fromJust $ getParseResult x
-
-consume :: ParserInfo c -> [String] -> c
-consume p = fromJust . getParseResult . execParserPure defaultPrefs p
+    consume (parseOptions pSpecs) input
+
+consume :: (Monad m) => ParserInfo a -> [String] -> m a
+consume p x = do
```

Comment:
Since this function isn’t monadic, I wouldn’t use `do`. Just use `let...in` instead.